conf/keycloak/test-realm.json

```diff
@@ -398,7 +398,7 @@
     "emailVerified" : true,
     "firstName" : "Dataverse",
     "lastName" : "Admin",
-    "email" : "dataverse-admin@mailinator.com",
+    "email" : "dataverse@mailinator.com",
```

Comment:
We talked about matching users from OIDC to Dataverse users multiple times at LIBIS. We feel like matching users by verified emails would be the best approach. The user with "dataverse-admin@malinator.com" does not exist in the dev env, so it did not work. Matching by email would mean that you can log in with google, facebook, whatever, as long as you use the same email.

Reply:
I reverted it and implemented the regular UserRecordIdentifier lookup, however, user is not known:
![image](https://github.com/user-attachments/assets/ea73906a-4d4d-410d-99b2-381440bfa9d2)

I have also implemented the redirect to the  first login page if user is not known:
![image](https://github.com/user-attachments/assets/85606fc1-03af-49b1-864d-9103bc5242cb)

After that, I can log in:
![image](https://github.com/user-attachments/assets/eaff340e-5371-4d1e-b399-30d27d9cd080)

Is it as it supposed to be? I have also noticed that oauth tokens are stored in the DB. The side effect of creating a new user after OIDC authentication is that it inserts one token there...

Reply:
I think I can just pass null as token data to prevent storing it in DB. I am not sure why we store them, I assume it is like the cache for bearer tokens? Something to investigate a little bit more.

Reply:
Passing `null` worked! The question of how to match users is still valid: verified email address sounds good to me. The current implementation uses the oauth way: user identifier based on the subject and provider ID. However, the existing users do not match for some reason, I am always redirected to the first time log in page when logging in for the first time. Maybe it is `oidc` as the provider ID for all OIDC providers? Or the preferred username i.s.o. the subject? Either way, it is easy to change in the current implementation. I would go either with the verified e-mail or user identifier based on the provider ID and subject (as it is implemented now). Maybe I am missing something here...